src/common/imageio.c

```diff
@@ -408,6 +411,13 @@ dt_imageio_retval_t dt_imageio_open_hdr(dt_image_t *img, const char *filename, d
   loader = LOADER_AVIF;
   if(ret == DT_IMAGEIO_OK || ret == DT_IMAGEIO_CACHE_FULL) goto return_label;
 #endif
+
+#ifdef HAVE_LIBHEIF
```

Comment:
We need a runtime detection if libheif has been compiled with heic support here!

Reply:
From what I see in libheif.h there is no chance to determine at runtime nor compile time if a specific extension is supported. And there are custom extensions, for example Canon uses **.HIF**.

Reply:
Please open a libheif bug report and request a way to find out if libheif is compiled with HEVC support.

Reply:
What exactly will happen right now if libheif was compiled without HEVC support?

Reply:
I think the standard load failure and "image corrupted" message presented to user in darkroom view...

Reply:
That's kindof runtime detection, maybe the message could be made more informative, like "recompile libheif with HEVC support to open this file"?.. But I agree that knowing beforehand is better.

Reply:
I see no real benefit in making support decision based on just a file extension. If libheif is available, all files with known extension should be passed to libheif. 
There is a better solution in libheif (I think). On open or decode, libheif returns specific error codes, for example:
````
 // Input file type is not supported.
  heif_error_Unsupported_filetype = 3,

  // Image requires an unsupported decoder feature.
  heif_error_Unsupported_feature = 4,
````
I had planned to handle such errors, but there is (afaik) no infrastructure in darktable to return specific error information. I can only return OK, CACHE_FULL or CORRUPTED from loader and most loaders I've seen return CORRUPTED in situations where the image is okay, but unable to handle them.
If it is possible to show a hint to the user during loading, please point me to how this can be done.
Keep in mind that darktable won't stop after a loader returns DT_IMAGEIO_CORRUPTED, it just try the next loader in chain, even if this loader is not related to the file extension.

Reply:
If libheif is unable to decode the file because of disabled HEVC support, a warning is printed to stderr.